Sources/Exporters/OpenTelemetryProtocol/common/EnvVarHeaders.swift

```diff
@@ -8,18 +8,37 @@ import OpenTelemetryApi
 
 /// Provides a framework for detection of resource information from the environment variable
 public struct EnvVarHeaders {
-    private static let otelAttributesEnv = "OTEL_EXPORTER_OTLP_HEADERS"
     private static let labelListSplitter = Character(",")
     private static let labelKeyValueSplitter = Character("=")
 
     ///  This resource information is loaded from the 
     ///  environment variable.
-    public static let attributes : [(String,String)]? = parseAttributes(rawEnvAttributes: ProcessInfo.processInfo.environment[otelAttributesEnv])
+    public static let attributes : [(String,String)]? = EnvVarHeaders.attributes()
+
+    public static func attributes(for rawEnvAttributes: String? = ProcessInfo.processInfo.environment["OTEL_EXPORTER_OTLP_HEADERS"]) -> [(String,String)]? {
```

Comment:
Signature here was changed to pass the raw environment attributes string in. It defaults to the value of the `OTEL_EXPORTER_OTLP_HEADERS` environment variable. The change made it possible to write a set of unit tests that cover this class.